src/FakeItEasy/CallOrderingContext.cs

```diff
@@ -0,0 +1,99 @@
+namespace FakeItEasy
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using FakeItEasy.Core;
+
+    /// <summary>
+    /// Represents a context to make assertions on the order of the calls made on a fake.
+    /// </summary>
+    public class CallOrderingContext : IDisposable

```

Comment:
I _do_ like the idea of exposing an interface…
I think with the `EditorBrowsable` and appropriate xml-docs, it's a good tradeoff. You've convinced me.